subprojects/core-api/src/main/java/org/gradle/api/artifacts/dsl/DependencyHandler.java

```diff
@@ -287,9 +287,8 @@
      * @param configurationName The name of the configuration.
      * @param dependencyNotation The dependency notation, in one of the notations described above.
      * @param configureClosure The closure to use to configure the dependency.
-     * @return The dependency.
+     * @return The dependency, or null if dependencyNotation is a provider.
      */
-    @Nullable
```

Comment:
Should it be documented somewhere that this change should be applied in 8.0?

Reply:
Yes, please open an issue

Reply:
I filed https://github.com/gradle/gradle/issues/18908.